Automattic/wp-calypso
# Purchases: Update the cancellation confirmation screen for domains

Reimplement the "confirm cancel puchase" form for domains on the client.
Previously the form was returned by the API and injected into Calypso.

#### Testing Instructions
- `git fetch -a && git checkout update/confirm-cancel`
- Purchase a .live domain from Calypso using the sandbox store
- Go to http://calypso.localhost:3000/purchases
- Find your domain purchase and click on it.
- The `Cancel Domain and Refund` option should be displayed.
- Click on it then click on `Cancel Domain and Refund`
- Ensure you land on http://calypso.localhost:3000/purchases/{SITE_ID}/{PURCHASE_ID}/confirm-cancel-domain
- Choose a reason for canceling your domain, check the checkbox (eventually enter a message in the textarea) and click on `Cancel Domain`.
- Verify that you are redirected to http://calypso.localhost:3000/purchases and that the following success notice is displayed: 
"[domain] was successfully cancelled and refunded."
- Verify that the domain no longer appears in the list of purchases

#### Reviews
